Assignment4/drachodran-e22a348/task07.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/15K7P-709udJAzhRvqiMLB2RKHD8m5xks
+
+**Task 07: Querying RDF(s)**
+"""
+
+#!pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+from rdflib.plugins.sparql import prepareQuery
+
+ns = Namespace("http://somewhere#")
+q1 = prepareQuery("""
+  SELECT ?personSubclass WHERE {
+    ?personSubclass rdfs:subClassOf ns:Person
+  }
+""",
+  initNs = { "ns": ns, "rdfs": RDFS }
+  )
+# Visualize the results
+
+for r in g.query(q1):
+  print(r)
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+
+q2 = prepareQuery('''
+  SELECT ?individuals WHERE {
+    {
+      ?individuals rdf:type ns:Person.
+    } 
+    UNION 
+    {
+      ?subclasses rdfs:subClassOf* ns:Person.
+      ?individuals rdf:type ?subclasses
+    }
+  }
+''',
+  initNs = { "ns": ns, "rdfs": RDFS, "rdf": RDF}
+  )
+for t in g.query(q2):
+  print(t)
+# Visualize the results
+
+"""**TASK 7.3: List all individuals of "Person" and all their properties including their class with RDFLib and SPARQL**
+
+"""
+
+from rdflib.plugins.sparql import prepareQuery
+
+ns = Namespace("http://somewhere#")
+q3 = prepareQuery('''
+  SELECT ?individuals ?properties WHERE {
+    {
+      ?individuals rdf:type ns:Person.
+      ?individuals ?properties ?o.
+      ?properties rdf:type rdf:Property
+    }
+     UNION 
+    {
+      ?subclasses rdfs:subClassOf* ns:Person.
+      ?individuals rdf:type ?subclasses.
+      ?individuals ?properties ?o.
+    }
+  }
+  ''', initNs = { "ns": ns, "rdf": RDF, "rdfs": RDFS }
+  )
+for y in g.query(q3):
+  print(y)
+

```

Comment:
You are still missing the exercises 7.1..7.3 using rdf lib. They are solved for SPARQL only

Reply:
Hi Dani,

it should be OK now.